maven-plugin-report-plugin/src/main/java/org/apache/maven/plugin/plugin/report/RequirementsHistory.java

```diff
@@ -72,16 +73,23 @@ public String toString() {
         return sb.toString();
     }
 
+    public static RequirementsHistory discoverRequirements(MavenProject project) {
+        RequirementsHistory req = new RequirementsHistory();
+        req.version = project.getVersion();
+        req.jdk = discoverJdkRequirement(project, null);
+        req.maven = discoverMavenRequirement(project, null);
+        return req;
+    }
     /**
      * Tries to determine the Maven requirement from either the plugin descriptor or (if not set) from the
      * Maven prerequisites element in the POM.
      *
      * @param project      not null
-     * @param pluginDescriptor the plugin descriptor (not null)
+     * @param pluginDescriptor the plugin descriptor (can be null)
      * @return the Maven version or null if not specified
      */
     public static String discoverMavenRequirement(MavenProject project, PluginDescriptor pluginDescriptor) {
-        if (StringUtils.isNotBlank(pluginDescriptor.getRequiredMavenVersion())) {
+        if (pluginDescriptor != null && StringUtils.isNotBlank(pluginDescriptor.getRequiredMavenVersion())) {
```

Comment:
How can it be suddenly `null`?

Reply:
because it was until now used only with current release, where the descriptor just had been generated
but now it is used with history, and I did not take time to extract generated content from Maven Central: I chose to just add code to not choke on null value

Reply:
I see, makes sense.